src/interface/sub_98199c.c

```diff
@@ -0,0 +1,158 @@
+	rct_g1_element *g1Element = gfx_get_g1_element(image_element);
+
+	rct_xyz16 coord_3d = {
+		.x = x_offset,
+		.y = y_offset,
+		.z = z_offset
+	};
+
+	switch (rotation) {
+		case 0:
+			rotate_map_coordinates(&coord_3d.x, &coord_3d.y, 0);
+			break;
+		case 1:
+			rotate_map_coordinates(&coord_3d.x, &coord_3d.y, 3);
+			break;
+		case 2:
+			rotate_map_coordinates(&coord_3d.x, &coord_3d.y, 2);
+			break;
+		case 3:
+			rotate_map_coordinates(&coord_3d.x, &coord_3d.y, 1);
+			break;
+	}
+	coord_3d.x += RCT2_GLOBAL(0x9DE568, sint16);
+	coord_3d.y += RCT2_GLOBAL(0x9DE56C, sint16);
+
+	rct_xy16 map = coordinate_3d_to_2d(&coord_3d, rotation);
+
+	ps->x = map.x;
+	ps->y = map.y;
+
+	int left = map.x + g1Element->x_offset;
+	int bottom = map.y + g1Element->y_offset;
+
+	int right = left + g1Element->width;
+	int top = bottom + g1Element->height;
+
+	RCT2_GLOBAL(0xF1AD1C, uint16) = left;
+	RCT2_GLOBAL(0xF1AD1E, uint16) = bottom;
+
+	rct_drawpixelinfo *dpi = RCT2_GLOBAL(0x140E9A8, rct_drawpixelinfo*);
+
+	if (right <= dpi->x) return false;
+	if (top <= dpi->y) return false;
+	if (left >= (dpi->x + dpi->width)) return false;
+	if (bottom >= (dpi->y + dpi->height)) return false;
+
+	rct_xyz16 boundBox = {
+		.x = bound_box_length_x,
+		.y = bound_box_length_y,
+		.z = bound_box_length_z
+	};
+
+	rct_xyz16 boundBoxOffset = {
+		.x = bound_box_offset_x,
+		.y = bound_box_offset_y,
+		.z = bound_box_offset_z
+	};
+
+	// Unsure why rots 1 and 3 need to swap
+	switch (rotation) {
+		case 0:
+			boundBox.x--;
+			boundBox.y--;
+			rotate_map_coordinates(&boundBoxOffset.x, &boundBoxOffset.y, 0);
+			rotate_map_coordinates(&boundBox.x, &boundBox.y, 0);
+			break;
+		case 1:
+			boundBox.x--;
+			rotate_map_coordinates(&boundBoxOffset.x, &boundBoxOffset.y, 3);
+			rotate_map_coordinates(&boundBox.x, &boundBox.y, 3);
+			break;
+		case 2:
+			rotate_map_coordinates(&boundBox.x, &boundBox.y, 2);
+			rotate_map_coordinates(&boundBoxOffset.x, &boundBoxOffset.y, 2);
+			break;
+		case 3:
+			boundBox.y--;
+			rotate_map_coordinates(&boundBox.x, &boundBox.y, 1);
+			rotate_map_coordinates(&boundBoxOffset.x, &boundBoxOffset.y, 1);
+			break;
+	}
+
+	ps->attached_x_end = boundBox.x + boundBoxOffset.x + RCT2_GLOBAL(0x009DE568, sint16);
+	ps->attached_z = boundBoxOffset.z;
+	ps->attached_z_end = boundBox.z + boundBoxOffset.z;
+	ps->attached_y_end = boundBox.y + boundBoxOffset.y + RCT2_GLOBAL(0x009DE56C, sint16);
+	ps->var_1A = 0;
+	ps->attached_x = boundBoxOffset.x + RCT2_GLOBAL(0x009DE568, sint16);
+	ps->attached_y = boundBoxOffset.y + RCT2_GLOBAL(0x009DE56C, sint16);
+	ps->attached_ps = NULL;
+	ps->var_20 = NULL;
+	ps->sprite_type = RCT2_GLOBAL(RCT2_ADDRESS_PAINT_SETUP_CURRENT_TYPE, uint8);
+	ps->var_29 = RCT2_GLOBAL(0x9DE571, uint8);
+	ps->map_x = RCT2_GLOBAL(0x9DE574, uint16);
+	ps->map_y = RCT2_GLOBAL(0x9DE576, uint16);
+	ps->mapElement = RCT2_GLOBAL(0x9DE578, rct_map_element*);
+
+	paint_struct *edi = RCT2_GLOBAL(0xF1AD28, paint_struct*);
```

Comment:
edi is the previously placed paint_struct. By attaching a var_20 you prevent attached_ps in edi painting and force var_20 to be drawn next. var_20 is a full ps (can have attachments.)